src/PureScript/Backend/Optimizer/CoreFn.purs

```diff
@@ -144,6 +145,12 @@ data Prop a = Prop String a
 
 derive instance functorProp :: Functor Prop
 
+instance Comonad Prop where
+  extract (Prop _ a) = a
+
+instance Extend Prop where
+  extend f p@(Prop k _) = Prop k (f p)
+
```

Comment:
This instances do not feel necessary to me. There is already `propKey` and `propValue` got getters, and it's not obvious to me that `extend` is going to lead to more straightforward code.